filebeat/input/journald/input.go

```diff
@@ -303,3 +331,25 @@ func (r *readerAdapter) Next() (reader.Message, error) {
 
 	return m, nil
 }
+func getJournaldVersion() (int, error) {
+	cmd := exec.CommandContext(context.Background(), "journalctl", "--version")
```

Comment:
I can live with this if there isn't another way, but running an external command is unexpected, it means that whatever runs filebeat has to have permissions to run journalctl, and we are tied to the output formatting of journalctl.

Is there any way you could load one of the systemd libraries and get the version number that way?

Reply:
+1, I linked https://github.com/systemd/systemd/issues/22654 in Slack separately.

Reply:
Thanks Folks!

I started looking into `pkg-config`, the file is well defined, however it's location is not :sob:

The [documentation](https://linux.die.net/man/1/pkg-config) specifies a relative path to it's configuration: `prefix/lib/pkgconfig`. On Archlinux and Ubuntu the Systemd file is at `/usr/share/pkgconfig/systemd.pc`, but on Amazon Linux 2 (the VM we have in Beats) it's at `/usr/lib64/pkgconfig/systemd.pc`.

The reliable way to find out where `pkg-config` will look for the files is to call `pkg-config --variable pc_path pkg-config`, which brings us back to the `fork` syscall problem :man_facepalming: 

I see two options here:
1. Look all the distros we officially support and hardcode the paths they use to store pkg-config files. We currently support just under 20 distros/versions for the latest Filebeat
2. Use D-Bus

I'll look into the D-Bus option.

Reply:
> I see two options here:
> 
>     1. Look all the distros we officially support and hardcode the paths they use to store pkg-config files. We currently support just under 20 distros/versions for the latest Filebeat
> 
>     2. Use D-Bus
> 
> 
> I'll look into the D-Bus option.

D-Bus seems like a good idea.  If that doesn't work, it might be worth checking to see if there are functions present in the newer versions of the library, that aren't in the older.  You could just try to dlopen & dlsym to try and load those functions, and use that for an indicator of what version you are on.

Reply:
I'm gonna resolve this conversation and start a new one about D-Bus